src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,222 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.lib.exception.DataProcessingException;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createCarStatement
+                        = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setString(1, car.getModel());
+            createCarStatement.setObject(2, car.getManufacturer().getId());
+            createCarStatement.executeUpdate();
+            ResultSet resultSet = createCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create car. " + car + " ",
+                    throwable);
+        }
+        insertDriver(car);
+        return car;
+    }
+
+    @Override
+    public Car get(Long id) {
+        String getRequest = "SELECT  c.id AS cars_id, model, m.id AS manufacturer_id, m.name \n"
+                + "FROM CARS c\n"
+                + "JOIN manufacturers m\n"
+                + "ON c.id = m.id\n"
+                + "WHERE c.id = ?;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement =
+                        connection.prepareStatement(getRequest)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                getCars(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can't insert manufacturer "
+                    + id, e);
+        }
+        if (car != null) {
+            car.setDriver(getDriverForCar(id));
+        }
+        return Optional.ofNullable(car).get();
```

Comment:
```suggestion
        return Optional.ofNullable(car);
```
there is no point of Optional then. Besides, DAO should return Optional